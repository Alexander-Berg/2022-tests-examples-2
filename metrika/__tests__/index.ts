import { remap } from '../normalize';

describe('remap', () => {
    it('works', () => {
        const result = remap(
            { yandexEmployeeLogin: '13' },
            { 13: 'loginFromStaff' },
        );

        expect(result).toEqual({ yandexEmployeeLogin: 'loginFromStaff' });
    });

    it('skips keys if they not presented in raw object', () => {
        const result = remap({ yandexEmployeeLogin: '13' }, {});

        expect(result).toEqual({});
    });

    it('skips keys if they not presented in mapping object', () => {
        const result = remap({}, { 13: 'loginFromStaff' });

        expect(result).toEqual({});
    });

    it('mixed case', () => {
        const result = remap(
            { someOtherField: '42' },
            { 13: 'loginFromStaff' },
        );

        expect(result).toEqual({});
    });
});
