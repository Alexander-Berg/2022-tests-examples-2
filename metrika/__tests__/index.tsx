import * as React from 'react';

import { render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Highlight, Substr } from '..';
configure({ adapter: new Adapter() });

describe('Highlight', () => {
    it('работает по строке', () => {
        const container = render(
            <div>
                <Highlight pattern="екс">текст</Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с вложенными строками', () => {
        const container = render(
            <div>
                <Highlight pattern="екс">
                    <span>текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с несколькими вложенными строками', () => {
        const container = render(
            <div>
                <Highlight pattern="екс">
                    <span>текст</span>
                    <span>текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с несколькими паттернами через пробел', () => {
        const container = render(
            <div>
                <Highlight pattern="ако екс">
                    <span>какой-то текст</span>
                    <span>какой-то текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с паттернами через массив', () => {
        const container = render(
            <div>
                <Highlight pattern={['ако', 'екс']}>
                    <span>какой-то текст</span>
                    <span>какой-то текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с подсветкой фона', () => {
        const container = render(
            <div>
                <Highlight pattern={['ако', 'екс']} highlight="background">
                    <span>какой-то текст</span>
                    <span>какой-то текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает позиционная подсветка', () => {
        const container = render(
            <div>
                <Substr match={{ start: 1, end: 3 }}>Here comes Johny!</Substr>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
});
