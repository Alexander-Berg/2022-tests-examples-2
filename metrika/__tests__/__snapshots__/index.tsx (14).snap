// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty size m 1`] = `
.c0 {
  display: inline-block;
  font-weight: normal;
}

.c0 .c1 {
  font-size: 15px;
  line-height: 18px;
}

<div
  class="c0"
/>
`;

exports[`empty size s 1`] = `
.c0 {
  display: inline-block;
  font-weight: normal;
}

.c0 .c1 {
  font-size: 13px;
  line-height: 15px;
}

<div
  class="c0"
/>
`;

exports[`with main and secondary size m 1`] = `
.c2 {
  color: #b8bec3;
  font-size: 13px;
  line-height: 15px;
}

.c3 .c1 {
  font-size: 13px;
  line-height: 15px;
}

.c0 {
  display: inline-block;
  font-weight: normal;
}

.c0 .c1 {
  font-size: 15px;
  line-height: 18px;
}

<div
  class="c0"
>
  <div
    class="c1 "
  >
    anything
  </div>
  <div
    class="c2"
  >
    anything
  </div>
</div>
`;

exports[`with main and secondary size s 1`] = `
.c2 {
  color: #b8bec3;
  font-size: 13px;
  line-height: 15px;
}

.c0 {
  display: inline-block;
  font-weight: normal;
}

.c0 .c1 {
  font-size: 13px;
  line-height: 15px;
}

<div
  class="c0"
>
  <div
    class="c1 "
  >
    anything
  </div>
  <div
    class="c2"
  >
    anything
  </div>
</div>
`;

exports[`with main size m 1`] = `
.c2 .c1 {
  font-size: 13px;
  line-height: 15px;
}

.c0 {
  display: inline-block;
  font-weight: normal;
}

.c0 .c1 {
  font-size: 15px;
  line-height: 18px;
}

<div
  class="c0"
>
  <div
    class="c1 "
  >
    anything
  </div>
</div>
`;

exports[`with main size s 1`] = `
.c0 {
  display: inline-block;
  font-weight: normal;
}

.c0 .c1 {
  font-size: 13px;
  line-height: 15px;
}

<div
  class="c0"
>
  <div
    class="c1 "
  >
    anything
  </div>
</div>
`;

exports[`with secondary size m 1`] = `
.c1 {
  color: #b8bec3;
  font-size: 13px;
  line-height: 15px;
}

.c0 {
  display: inline-block;
  font-weight: normal;
}

.c0 .c2 {
  font-size: 15px;
  line-height: 18px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    anything
  </div>
</div>
`;

exports[`with secondary size s 1`] = `
.c1 {
  color: #b8bec3;
  font-size: 13px;
  line-height: 15px;
}

.c0 {
  display: inline-block;
  font-weight: normal;
}

.c0 .c2 {
  font-size: 13px;
  line-height: 15px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    anything
  </div>
</div>
`;
